thermalnetwork/network.py

```diff
@@ -628,42 +630,75 @@ def size(self, output_path: Path, throw: bool = True):
 
         :return: None
         """
 
         # Load the existing system parameters
-        sys_params: dict = json.loads(system_parameter_path.read_text())
-        ghe_params: list = sys_params["district_system"]["fifth_generation"]["ghe_parameters"]["ghe_specific_params"]
-        for ghe in ghe_params:
+        sys_params = load_json(system_parameter_path)
+        dist_sys_params = sys_params["district_system"]["fifth_generation"]
+        ghe_params = dist_sys_params["ghe_parameters"]
+        ghe_specific_params: list = ghe_params["ghe_specific_params"]
+        pipe_params = dist_sys_params["horizontal_piping_parameters"]
+        pump_params = dist_sys_params["central_pump_parameters"]
+
+        max_num_boreholes = 0
+        bh_vol_flow = ghe_params["design"]["flow_rate"]
+
+        # update ghe parameters
+        for ghe in ghe_specific_params:
             ghe_id = ghe["ghe_id"]
             summary_data_path = output_directory_path / ghe_id / "SimulationSummary.json"
 
             if not summary_data_path.exists():
                 logger.error(f"Error sizing GHE: {ghe_id}")
 
             # Get the new data from the GHEDesigner output
-            ghe_data: dict = json.loads(summary_data_path.read_text())["ghe_system"]
-            for ghe_sys_params in ghe_params:
+            ghe_data = load_json(summary_data_path)["ghe_system"]
+            for ghe_sys_params in ghe_specific_params:
                 if ghe_sys_params["ghe_id"] == ghe_id:
                     # Update system parameters dict with the new values
-                    ghe_sys_params["borehole"]["length_of_boreholes"] = ghe_data["active_borehole_length"]["value"]
-                    ghe_sys_params["borehole"]["number_of_boreholes"] = ghe_data["number_of_boreholes"]
-        with open(system_parameter_path, "w") as sys_param_file:
-            json.dump(sys_params, sys_param_file, indent=2)
-            # Restore the trailing newline
-            sys_param_file.write("\n")
+                    bh_len = ghe_data["active_borehole_length"]["value"]
+                    num_boreholes = ghe_data["number_of_boreholes"]
+                    ghe_sys_params["borehole"]["length_of_boreholes"] = bh_len
+                    ghe_sys_params["borehole"]["number_of_boreholes"] = num_boreholes
+
+                    max_num_boreholes = max(num_boreholes, max_num_boreholes)
+
+        # update pump and pipe params
+        network_design_vol_flow = lps_to_cms(bh_vol_flow) * max_num_boreholes
+        network_pipe = Pipe(
+            dimension_ratio=pipe_params["diameter_ratio"],
+            length=self.total_network_pipe_length,
+            fluid_type=ghe_params["fluid"]["fluid_name"],
+            fluid_concentration=ghe_params["fluid"]["concentration_percent"],
+            fluid_temperature=ghe_params["fluid"]["temperature"],
+        )
+
+        # we should expose this to the user at some point
+        design_pressure_loss_per_length = 300  # Pa/m
```

Comment:
I realized Modelica has a similar parameter `dp_length_nominal` which is used by the horizontal piping model. We should expose this in the sys_params and use it both here and in the Modelica model so they align with each other.
![image](https://github.com/user-attachments/assets/53510295-537a-40e7-bd58-ae848a5aceef)

Reply:
That's a great find. Yes, I believe that is the right move here.

Reply:
I opened an GMT issue here: https://github.com/urbanopt/geojson-modelica-translator/issues/696.